### PR TITLE
Add E2E tests for subnet import

### LINKED ISSUE #224: Add E2E tests and flags for subnet import
We need to test both the file and repo import paths. We need to add flags to facilitate this.

### PR DESCRIPTION
**Pull request by dasconnor**
Added E2E tests for the `subnet import` command. To do so, I needed to add flags to the command. I also moved all of the test cases in the e2e test suite into a `testcases` folder.

Closes #224

### CODE REVIEW

File: tests/e2e/utils/constants.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ package utils
 
 const (
 	baseDir               = ".avalanche-cli"
+	baseAPMDir            = ".apm"
```
Comment (felipemadero) [inline tests/e2e/utils/constants.go:8]: can't we use constants in `pkg/constants/constants` for this?

File: tests/e2e/utils/helpers.go [hunk 2]
```diff
@@ -70,6 +79,19 @@ func SubnetConfigExists(subnetName string) (bool, error) {
 	return genesisExists && sidecarExists, nil
 }
 
+func APMConfigExists(subnetName string) (bool, error) {
```
Comment (felipemadero) [inline tests/e2e/utils/helpers.go:82]: this is check for subnet sidecar to exist. maybe can be renamed, and also reused in the other functions here

File: cmd/subnetcmd/import.go [hunk 3]
```diff
@@ -37,6 +43,24 @@ with the same name. To allow overwrites, provide the --force flag.`,
 		false,
 		"overwrite the existing configuration if one exists",
 	)
+	cmd.Flags().StringVar(
```
Comment (felipemadero) [inline cmd/subnetcmd/import.go:46]: probably we need to improve the Long description, and maybe add a comment at the start of the function. be clear that it either imports from a file or from apm

File: cmd/subnetcmd/import.go [hunk 4]
```diff
@@ -114,30 +140,58 @@ func importFromAPM() error {
...
-	installedRepos = append(installedRepos, customRepo)
 
-	promptStr := "What repo would you like to import from"
-	repoAlias, err := app.Prompt.CaptureList(promptStr, installedRepos)
-	if err != nil {
-		return err
+	if repoOrURL != "" {
+		fmt.Println("Looking for", repoOrURL)
+		for _, installedRepo := range installedRepos {
+			fmt.Println("checking against", installedRepo)
```
Comment (felipemadero) [inline cmd/subnetcmd/import.go:151]: ux?

File: tests/e2e/commands/subnet.go [hunk 5]
```diff
@@ -399,3 +399,90 @@ func SimulateJoinPublicly(
...
+	cmd := exec.Command(
+		CLIBinary,
+		SubnetCmd,
+		"import",
+		"--repo",
+		repoAlias,
+		"--subnet",
+		subnetName,
+	)
+	fmt.Println(cmd.String())
```
Comment (felipemadero) [inline tests/e2e/commands/subnet.go:424]: maybe remove this

File: tests/e2e/testcases/apm/suite.go [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+		utils.DeleteAPMBin(vmid2)
+		// TODO same as above
+		utils.RemoveAPMRepo()
+	})
+
+	ginkgo.It("can import from avalanche-core", func() {
+		repo := "ava-labs/avalanche-plugins-core"
+		subnet := "wagmi"
+
+		commands.ImportSubnetConfig(repo, subnet)
```
Comment (felipemadero) [inline tests/e2e/testcases/apm/suite.go:53]: subnet1?

File: tests/e2e/testcases/apm/suite.go [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	})
+
+	ginkgo.It("can import from url", func() {
+		branch := "master"
+		subnet := "spaces"
+
+		commands.ImportSubnetConfigFromURL(testRepo, branch, subnet)
```
Comment (felipemadero) [inline tests/e2e/testcases/apm/suite.go:60]: subnet2?

File: tests/e2e/utils/constants.go (hunk 1 above)
Comment (dasconnor) [reply tests/e2e/utils/constants.go]: fixed

File: tests/e2e/utils/helpers.go (hunk 2 above)
Comment (dasconnor) [reply tests/e2e/utils/helpers.go:97]: I reworked this so sidecarExists and genesisExists are separate functions and this just wraps that function.

File: cmd/subnetcmd/import.go (hunk 3 above)
Comment (dasconnor) [reply cmd/subnetcmd/import.go:49]: Reworked the description

File: cmd/subnetcmd/import.go (hunk 4 above)
Comment (dasconnor) [reply cmd/subnetcmd/import.go]: removed

File: tests/e2e/commands/subnet.go (hunk 5 above)
Comment (dasconnor) [reply tests/e2e/commands/subnet.go]: removed

File: tests/e2e/testcases/apm/suite.go (hunk 6 above)
Comment (dasconnor) [reply tests/e2e/testcases/apm/suite.go]: done

File: tests/e2e/testcases/apm/suite.go (hunk 7 above)
Comment (dasconnor) [reply tests/e2e/testcases/apm/suite.go]: done

**Review by felipemadero — APPROVED** [review]
LGTM

**Pull request merged**